apps/base-docs/docs/building-with-base/guides/cross-chain-with-ccip.md

```diff
@@ -0,0 +1,615 @@
+---
+title: Chainlink CCIP
+slug: /guides/cross-chain-with-ccip
+description: A guide on using Chainlink CCIP to perform cross-chain messaging and token transfers from Base Goerli testnet to Optimism Goerli testnet.
+keywords:
+  [
+    Cross-chain,
+    Omni-chain,
+    Crosschain,
+    OmniChain,
+    Chainlink,
+    Chainlink CCIP,
+    CCIP,
+    cross-chain messaging,
+    transfer tokens across chains,
+  ]
+---
+
+# Sending messages and tokens from Base to other chains using Chainlink CCIP
+
+This page will guide you through the process of sending messages and tokens from a Base smart contract to another smart contract on a different chain using Chainlink's Cross-chain Interoperability Protocol (CCIP).
+
+---
+
+## Objectives
+
+By the end of this guide you should be able to do the following:
+
+- Set up a smart contract project for Base using Foundry
+- Install Chainlink CCIP as a dependency
+- Use Chainlink CCIP within your smart contract to send messages and/or tokens to contracts on other different chains
+- Deploy and test your smart contracts on Base testnet
+
+:::info
+
+Chainlink CCIP is in an “Early Access” development stage, meaning some of the functionality described within this guide is under development and may change in later versions.
+
+:::
+
+---
+
+## Prerequisites
+
+### Foundry
+
+This guide requires you to have Foundry installed.
+
+- From the command-line (terminal), run: `curl -L https://foundry.paradigm.xyz | bash`
+- Then run `foundryup`, to install the latest (nightly) build of Foundry
+
+For more information, see the Foundry Book [installation guide](https://book.getfoundry.sh/getting-started/installation).
+
+### Coinbase Wallet
+
+In order to deploy a smart contract, you will first need a wallet. You can create a wallet by downloading the Coinbase Wallet browser extension.
+
+- Download [Coinbase Wallet](https://chrome.google.com/webstore/detail/coinbase-wallet-extension/hnfanknocfeofbddgcijnmhnfnkdnaad?hl=en)
+
+### Wallet funds
+
+For this guide you will need to fund your wallet with both ETH and LINK on Base Goerli and Sepolia Goerli.
+
+The ETH is required for covering gas fees associated with deploying smart contracts to the blockchain, and the LINK token is required to pay for associated fees when using CCIP.
+
+- To fund your wallet with ETH on Base Goerli, visit a faucet listed on the [Base Faucets](https://docs.base.org/tools/network-faucets) page.
+- To fund your wallet with ETH on Optimism Goerli, visit a faucet listed on the [Optimism Faucets](https://docs.optimism.io/builders/tools/faucets) page.
+- To fund your wallet with LINK, visit the [Chainlink Faucet](https://faucets.chain.link/base-testnet).
+
+:::info
+
+If you are interested in building on Mainnet, you will need to [apply for Chainlink CCIP mainnet access](https://chainlinkcommunity.typeform.com/ccip-form?#ref_id=ccip_docs).
+
+:::
+
+---
+
+## What is Chainlink CCIP?
```

Comment:
I think we need a little more of an overview of how this is working.  Are we trusting Chainlink?  Or is this process provable.  I'm only just starting to look into cross-chain messaging myself, but my impression was that to do Base to OP, you needed to go through mainnet, which would take 7 days up and 2-3 minutes down.  But it looks like this is doing it directly in 2 or 3 minutes?  How?

Reply:
I'll add more detail on CCIP's architecture and how it works under the hood.

Reply:
Added high-level architecture/concepts section.